### DISCUSSION TITLE
Using tap with pipes in DSL2

### DISCUSSION THREAD

**Discussion by glormph** [nextflow-io/nextflow#3919] [Q&A]
Hi, I have tried and failed to use `tap` in DSL2 with pipes:
```
ch1
| SomeProcess
| tap { ch2 }
| AnotherProcess
```
This gets me an error message, `No such variable: ch2`. I have not been able to figure out if this is intended or not, since `.tap { ch2 }` works fine. Not a huge problem and easy enough to work around, but I do like the pipe syntax a lot :)

**Comment by bentsherman** [comment]
You are right, this is a bug

**Reply by bentsherman** [reply]
Converted to issue: #3970
